parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -868,6 +868,24 @@ providers:
     apiVersion: credentialprovider.kubelet.k8s.io/v1
     args:
       - /etc/kubernetes/azure.json
```

Comment:
The only change is here?

consider add a 'CREDENTIAL_PROVIDER_ADDITIONAL_ARG' var and reuse whole file content?

Reply:
We need to add mcr.microsoft.com to the image match list as well.

https://github.com/Azure/AgentBaker/pull/5836/files#diff-4f9504893023c819ba00993d7db2a80f9883ef64f249e6e61b06d18094f3d425R883

Reply:
let's keep three `if` statements in this PR. @mainred will consider refactoring it later.